dnf/dnf-main.c

```diff
@@ -283,7 +287,14 @@ main (int   argc,
 
       dnf_transaction_set_flags (txn, flags);
     }
-
+  if (!show_help && opt_best)
```

Comment:
I prefer this code below test and error message. And move it up into `if (!show_help)`:
```
      if (opt_best && opt_nobest)
        {
          error = g_error_new_literal(G_OPTION_ERROR,
                                      G_OPTION_ERROR_BAD_VALUE,
                                      "Argument --nobest is not allowed with argument --best");
          goto out;
        }
      if (opt_best)
        {
          dnf_context_set_best(TRUE);
        }
      else if (opt_nobest)
        {
          dnf_context_set_best(FALSE);
        }
```